packages/components/src/InputNumber/InputNumber.test.tsx

```diff
@@ -199,6 +199,70 @@ test("allows custom validation", async () => {
   });
 });
 
+test("it should call the custom validate function if provided", async () => {
+  const validationHandler = jest.fn();
+  const expectedValidationValue = Math.floor(
+    Math.random() * Number.MAX_SAFE_INTEGER,
+  );
+  const { getByLabelText } = render(
+    <InputNumber
+      value={expectedValidationValue}
+      placeholder="Custom validation function"
+      validations={{
+        validate: validationHandler,
+      }}
+    />,
+  );
+
+  const input = getByLabelText("Custom validation function");
+
+  input.focus();
+  input.blur();
+
+  await waitFor(() => {
+    expect(validationHandler).toHaveBeenCalled();
+    // expect(validationHandler).toHaveBeenCalledWith(expectedValidationValue);
+    // Expected: 12
+    // Received: 12, {"generatedName--123e4567-e89b-12d3-a456-426655440063": 12}
```

Comment:
I'd like to do `toHaveBeenCalledWith` to be more precise about what's going on, but I haven't been able to figure out how to get Jest to match a string within `toHaveBeenCalledWith`. I'm asking the testing guild, but it won't change the code, so this can be reviewed.